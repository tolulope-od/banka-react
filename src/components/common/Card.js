import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ cardContent }) => {
  return <div className="account-stats-card">{cardContent}</div>;
};

Card.propTypes = {
  cardContent: PropTypes.object.isRequired
};

export default Card;
