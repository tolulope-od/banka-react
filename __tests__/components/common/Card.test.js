import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from '../../../src/components/common/Card';

const shallowSetup = () => {
  const props = {
    cardContent: {}
  };

  const enzymeWrapper = shallow(<Card {...props} />);

  return { props, enzymeWrapper };
};

describe('<Card /> Component', () => {
  const { props } = shallowSetup();

  it('Renders the Card component', () => {
    const wrapper = shallow(<Card {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
