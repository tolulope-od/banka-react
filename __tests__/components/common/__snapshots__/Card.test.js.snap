// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Component Renders the Card component 1`] = `
<div
  className="account-stats-card"
>
  <Component />
</div>
`;
